### PR TITLE
Modularize MTP benchmark orchestration and report validation

### PR DESCRIPTION
**Pull request by Gajesh2007**
The MTP benchmark launcher mixed cache discovery, artifact fingerprints, output safety, report validation, process supervision and CLI handling in 1,622 lines. Split those responsibilities into focused `scripts/mtp_benchmark` modules, retain the original launcher and helper imports, and consolidate repeated fixed/adaptive MTP evidence checks without changing refusal order or messages.

The entry point is now 429 lines. Total production Python grows by 76 lines for module imports and compatibility exports; the reduction is in duplicate validation/control flow and the size of the entry point. Artifact identities, no-download policy, process-group deadlines, output-directory protections, worker flags, schema, printed receipts and return codes are preserved.

Before

```mermaid
flowchart TD
  A[Existing CLI invocation] --> B[run-mtp-benchmark.py: parse and supervise]
  B --> C[Same file: private run directory and worker launch]
  C --> D[Same file: cache discovery and artifact fingerprints]
  D --> E[Supervised offline Swift test]
  E --> F[Same file: drift plus nested report and duplicated fixed/adaptive checks]
  F --> G[Accepted report or original failure receipt and exit code]
  B --> H[Same-file CPU self-test flags]
```

After

```mermaid
flowchart TD
  A[Same CLI flags and helper imports] --> B[run-mtp-benchmark.py: same supervisor and worker]
  B --> C[run_directory.py: same descriptor and inode protections]
  B --> D[artifacts.py: same offline discovery and fingerprints]
  C --> E[Same supervised offline Swift test]
  D --> E
  E --> F[report.py: launch provenance and opaque matrix evidence]
  F --> G[metrics.py: shared inactive/verifier policy and mode-specific work checks]
  G --> H[Same accepted report or failure receipt and exit code]
  B --> I[self_tests.py: same CPU self-test flags]
  J[Release Integrity CI and make benchmark-wrapper-test] --> K[Characterization tests with synthetic reports and mocked launches]
```

Validation: 73 wrapper tests pass (65 existing, 8 new); the new 8 pass against both original and refactored code; 1,000 extra report mutations agree exactly. Both CLI self-tests pass; docs lint passes 278 files. No model/GPU process was run. Based on `master`; the follow-up malformed-evidence fixes are isolated in #956.

---
<a href="https://app.blacksmith.sh/Layr-Labs/codesmith/d-inference/pr/901"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1791754768&installation_model_id=21733&pr_number=901&repository=Layr-Labs%2Fd-inference&return_to=https%3A%2F%2Fgithub.com%2FLayr-Labs%2Fd-inference%2Fpull%2F901&signature=2d0546719ea33f22e148ca95492df88d8c3db720698213ccd30eaac40ee74b3c"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by Gajesh2007** [comment]
Final validation for `121b70804831d6a0c50c95b6542d917f68ed3d54`:

- All ordinary PR CI checks, including provider and E2E integration tests, pass. The separate environment-gated E2E benchmark job remains waiting.
- Codex completed review of this head with a thumbs-up; no unresolved review threads remain.
- The five sibling refactors (#897, #898, #899, #900, #901) combine cleanly in a local validation branch based on GPT-OSS #894. Combined coordinator streaming/settlement race tests, testbed and targeted E2E CPU race tests pass. The final combination also passes all 73 benchmark-wrapper tests, docs lint (279 files), and whitespace checks.

Individual validation details and limitations are in each PR description, including the dedicated-M5 provider/live GPT-OSS checks and the unchanged baseline TypeScript diagnostics in #899. No merge or deployment was performed.

File: scripts/run-mtp-benchmark.py [hunk 1]
```diff
@@ -19,547 +19,42 @@
...
+if _SCRIPT_DIRECTORY not in sys.path:
+    sys.path.insert(0, _SCRIPT_DIRECTORY)
+
+from mtp_benchmark.constants import (
+    REPO_ROOT, PACKAGE_ROOT, DEFAULT_TARGET_ID, DEFAULT_ASSISTANT_ID, DEFAULT_TEST_FILTER,
+    REPORT_SCHEMA_VERSION, REPORT_NAME, LOG_NAME, SUPERVISOR_CONTRACT,
+    LEGACY_M5_INACTIVE_REASON_PREFIX, MAX_CACHE_ROOTS, MAX_FALLBACK_REPOSITORY_ENTRIES,
+    MAX_FALLBACK_REPOSITORIES, MAX_SNAPSHOT_ENTRIES, MAX_REF_BYTES, MAX_REPORT_BYTES,
+    PERFORMANCE_KEYS, HEX_DIGITS,
+)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply scripts/run-mtp-benchmark.py:34]: These imports intentionally preserve the original launcher’s importable helper interface after extraction, as described in the PR. Several names reported as unused here are called through the launcher by scripts/mtp_benchmark/tests/test_contract.py, including LEGACY_M5_INACTIVE_REASON_PREFIX, expected_coverage and validate_automatic_fixed_fallback. Removing individual exports would silently change that interface. The CLI uses the focused modules internally, and the existing characterization tests exercise the retained interface. Keeping these compatibility re-exports; this finding does not indicate dead implementation.

File: scripts/run-mtp-benchmark.py [hunk 2]
```diff
@@ -19,547 +19,42 @@
...
+from mtp_benchmark.artifacts import (
+    repo_cache_name, bounded_scandir, cache_roots, confined_regular_file,
+    repository_snapshot, resolve_cached_snapshot, validate_snapshot,
+    infer_huggingface_model_id, resolve_model_snapshot, sha256_file,
+    append_fingerprint_field, collect_nested_bit_overrides,
+    launch_effective_quantization_bits, artifact_facts,
+)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply scripts/run-mtp-benchmark.py:41]: These imports intentionally preserve the original launcher’s importable helper interface after extraction, as described in the PR. Several names reported as unused here are called through the launcher by scripts/mtp_benchmark/tests/test_contract.py, including LEGACY_M5_INACTIVE_REASON_PREFIX, expected_coverage and validate_automatic_fixed_fallback. Removing individual exports would silently change that interface. The CLI uses the focused modules internally, and the existing characterization tests exercise the retained interface. Keeping these compatibility re-exports; this finding does not indicate dead implementation.

File: scripts/run-mtp-benchmark.py [hunk 3]
```diff
@@ -19,547 +19,42 @@
...
+from mtp_benchmark.run_directory import (
+    open_directory, SecureRunDirectory, validate_leaf_name, write_all, read_bounded,
+)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply scripts/run-mtp-benchmark.py:44]: These imports intentionally preserve the original launcher’s importable helper interface after extraction, as described in the PR. Several names reported as unused here are called through the launcher by scripts/mtp_benchmark/tests/test_contract.py, including LEGACY_M5_INACTIVE_REASON_PREFIX, expected_coverage and validate_automatic_fixed_fallback. Removing individual exports would silently change that interface. The CLI uses the focused modules internally, and the existing characterization tests exercise the retained interface. Keeping these compatibility re-exports; this finding does not indicate dead implementation.

File: scripts/run-mtp-benchmark.py [hunk 4]
```diff
@@ -19,547 +19,42 @@
...
+from mtp_benchmark.metrics import (
+    observed_bucket, expected_mtp_expectation, inactive_reason_matches,
+    validate_zero_speculative_work, automatic_rectangular_cap, positive_cost_inputs,
+    positive_costs_within_cap, validate_automatic_fixed_fallback,
+    validate_automatic_adaptive_within_cap,
+)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply scripts/run-mtp-benchmark.py:50]: These imports intentionally preserve the original launcher’s importable helper interface after extraction, as described in the PR. Several names reported as unused here are called through the launcher by scripts/mtp_benchmark/tests/test_contract.py, including LEGACY_M5_INACTIVE_REASON_PREFIX, expected_coverage and validate_automatic_fixed_fallback. Removing individual exports would silently change that interface. The CLI uses the focused modules internally, and the existing characterization tests exercise the retained interface. Keeping these compatibility re-exports; this finding does not indicate dead implementation.

File: scripts/run-mtp-benchmark.py [hunk 5]
```diff
@@ -19,547 +19,42 @@
...
+from mtp_benchmark.report import (
+    parse_timestamp, recursively_present_keys, effective_quantization_bits,
+    expected_coverage, validate_report_artifact, validate_report,
+)
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply scripts/run-mtp-benchmark.py:54]: These imports intentionally preserve the original launcher’s importable helper interface after extraction, as described in the PR. Several names reported as unused here are called through the launcher by scripts/mtp_benchmark/tests/test_contract.py, including LEGACY_M5_INACTIVE_REASON_PREFIX, expected_coverage and validate_automatic_fixed_fallback. Removing individual exports would silently change that interface. The CLI uses the focused modules internally, and the existing characterization tests exercise the retained interface. Keeping these compatibility re-exports; this finding does not indicate dead implementation.

**Comment by Gajesh2007** [comment]
Consolidated into #1096, the combined tooling, benchmark and E2E refactor PR. Your latest head `d9f7ad0b95358bace724aefb29d7da49e24fcca9` is preserved in aggregate `9d2160bb370e8a0dbf3810e61a5e48fe66345651`, along with the relevant review fixes. Continue review and merge at https://github.com/Layr-Labs/d-inference/pull/1096 (after provider #1094 and web #1095). Closing this duplicate merge path; the source branch and review history are retained.
